### PR TITLE
REUSE compliance badge_v.0

### LINKED ISSUE #599: Make repository REUSE compliant
TBD around March 2023. Maybe after decision on re-licencing was made.

### PR DESCRIPTION
**Pull request by juli-a-ko**
# Closes #599

## Checklist

- [ ] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by ekatef** [comment]
@juli-a-ko amazing work! Thank you so much for the PR :)

**Comment by pz-max** [comment]
Cool! Just some notes that need to be checked:

- Looks like some csv got deleted by accident
- All Python scripts should be GPL3 licensed (not MIT)
- rst files need to be checked if the license note is not conflicting the documentation (make html to create the documentation locally)

**Comment by ekatef** [comment]
@pz-max thanks a lot for looking into it! We'll go point-by-point in the review round and then ask you for the second check. @juli-a-ko has done a very big work which currently needs to be carefully accommodated in the project

Max, there is also an additional question: it would be great to see the green reuse badge but it looks like we need to register the fork project on reuse. Not sure however if it wouldn't it be misuse of reuse infrastructure 🙂 What do you think?

**Comment by pz-max** [comment]
> Max, there is also an additional question: it would be great to see the green reuse badge but it looks like we need to register the fork project on reuse. Not sure however if it wouldn't it be misuse of reuse infrastructure slightly_smiling_face What do you think?

Sure, go ahead with registering the badge on the fork to test it... we can delete the for registration afterwards? If not, let's do the badge stuff afterwards on the main repo :)

**Comment by euronion** [comment]
This PR closes #599 if finished, right? :) Good work so far!

A few comments from my side as well:

* The `<year>` [does not need to be specified in the SPDX identifier](https://reuse.software/faq/#years-copyright), i.e. `reuse annotate --year="" ...` . I choose this with other recent options because I think it looks nicer and doesn't cause the urge to update it every year ;-)
* Some SPDX identifiers look broken (e.g. missing `#`) at beginning of line, `reuse lint` should complain about those
* Make sure to avoid redundant licence information, e.g. for `.github/*` via `.reuse/dep5` but also a licence specified in the individual files under `.github/` via an SPDX identifier
* Special attention is probably needed for the files in `scripts` which are based/copied from PyPSA-EUR and then partially adapted. E.g. `scripts/_helpers.py` contains mixed efforts from PyPSA-EUR and PyPSA-Earth and needs the appropriate CopyRightText as well as licence. It is worth noting that PyPSA-EUR underwent a [licence change in 2021](https://pypsa-eur.readthedocs.io/en/latest/release_notes.html#pypsa-eur-0-4-0-22th-september-2021) to MIT licence, but I think the old code fragments are from before that?
* @pz-max : Is it decided to go with GPLv3?
* We don't have a contributor agreement yet, in the meantime it could be helpful to include a checkbox in the [Pull Request template](https://github.com/pypsa-meets-earth/pypsa-earth/blob/main/.github/pull_request_template.md) like we do in [technology data](https://github.com/PyPSA/technology-data/blob/master/.github/pull_request_template.md), e.g.
```
[ ] I consent to the release of this PR's code under the GPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
```

**Comment by pz-max** [comment]
All comments/suggestions from @euronion are valid (as always). 
- Regarding mixed efforts. I think  we could mention "PyPSA-Earth and PyPSA-Eur authors" on all scripts to keep things simple
- Correct, GPLv3 is the current solution

**Comment by ekatef** [comment]
@euronion, @pz-max huge thanks for the comments!

Jonannes, that is perfect to have your expert opinion. Indeed, I absolutely agree with Max that it's worth to mention in all the scripts both PyPSA-Eur and PyPSA-Earth teams. A good point about the licensing consent!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# SPDX-FileCopyrightText: 2022-2023 PyPSA-Earth Authors
+#
+ SPDX-License-Identifier: CC0-1.0
```
Comment (ekatef) [inline .github/pull_request_template.md]: It looks like "#" is missed here

File: .gitignore [hunk 2]
```diff
@@ -31,6 +33,7 @@ resources/
 scripts/temp
 scripts/test.py
 test/tmp/
+doc/configtables
```
Comment (ekatef) [inline .gitignore]: Not sure we have to include csv-s and doc/configtables into gitignore

File: README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # PyPSA-Earth
-
+#
+SPDX-FileCopyrightText: 2022-2023 PyPSA-Earth Authors
```
Comment (ekatef) [inline README.md]: "#" seems to get lost

File: doc/configtables/atlite.csv
Comment (ekatef) [inline doc/configtables/atlite.csv:1]: It looks like configtables were removed by accident :)

**Review by ekatef — CHANGES_REQUESTED** [review]
@juli-a-ko, thanks a lot for the PR and sorry for taking a while to review it.

Have added few technical comments. Regarding checking rst files (source [reStructured Text](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) files used to build project documentation), will be happy to help in testing it :)

**Comment by pz-max** [comment]
Hey everyone, I am currently working a PR for this and took some inspiration from here. I will push a PR this night.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ekatef** [comment]
Hey @pz-max, thanks for notifying! Looking forward for your PR :)

**Comment by pz-max** [comment]
@juli-a-ko @ekatef thanks for great try with the REUSE stuff. 
I just opened a PR at #651 which might interest you.
Sorry for rushing that, I just had some time today and ws curious :)

**Pull request closed without merging**